inference-engine/src/vpu/graph_transformer/src/stages/tensor_iterator.cpp

```diff
@@ -344,7 +374,27 @@ void FrontEnd::parseTensorIterator(const Model& model, const ie::CNNLayerPtr& la
         const auto& bodyOutputs = tensorIterator->body.outputs;
         VPU_THROW_UNLESS(!bodyOutputs.empty(), "If there is no an output for Tensor Iterator's body, so there is no iteration in tensor iterator");
 
-        for (const auto& bodyOutput : bodyOutputs) {
+        for (std::size_t bodyOutputIdx = 0; bodyOutputIdx < bodyOutputs.size(); ++bodyOutputIdx) {
+            const auto& bodyOutput = bodyOutputs[bodyOutputIdx];
+
+            if (tensorIterator->params.count(s_condPort) && tensorIterator->GetParamAsUInt(s_condPort) == bodyOutputIdx) {
+                const auto& creator = getCreatorLayer(bodyOutput).lock();
+                if (!creator) {
+                    // ConstTransformer leaves constant without creator
+                    // Assume it's true
+                    continue;
+                }
+                VPU_THROW_UNLESS(isConst(bodyOutput), "Body execution condition must be constant true");
+
+                VPU_THROW_UNLESS(creator->blobs.size(), "Body execution condition constant must have one blob");
```

Comment:
the condition corresponds to the error message?

Reply:
`creator->blobs.size() == 1` ?

Reply:
Nice catch, thanks!

Reply:
Fixed